### PR TITLE
[Feature] Export data supports parquet format

### PR DESCRIPTION
**Pull request by xinghuayu007**
## Proposed changes

Currently, Doris only supports export data with CSV format. This PR extends the feature to support parquet format.

Usage:

for export to local file:

`export table table2 to "file:///root/doris/" properties ("format"="parquet", "schema"="required, int64, date; required, int32, siteid; required, int32, citycode; required, byte_array, username; required, int64, pv;", "parquet.version"="v2");`

If `format` is `parquet`,  then `schema` is required

for export to remote file:

`export table table2 to "hdfs://host/test_sql_prc_2019_02_19/" PROPERTIES ("schema"="required,int32,siteid;required,byte_array,varchar_test", "schema"="required, int64, date; required, int32, siteid; required, int32, citycode; required, byte_array, username; required, int64, pv;", "parquet.version"="v2") with broker hdfs_broker ("hadoop.security.authentication" = "kerberos",     "kerberos_principal" = "account",     "kerberos_keytab_content" = "base64");`

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If these changes need document changes, I have updated the document
- [x] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/analysis/LoadStmt.java [hunk 1]
```diff
@@ -95,6 +95,7 @@
     public static final String KEY_IN_PARAM_LINE_DELIMITER = "line_delimiter";
     public static final String KEY_IN_PARAM_PARTITIONS = "partitions";
     public static final String KEY_IN_PARAM_FORMAT_TYPE = "format";
+    public static final String KEY_IN_PARAM_FILE_PROPERTIES = "file_properties";
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/LoadStmt.java:98]: Seems useless?

File: gensrc/thrift/DataSinks.thrift [hunk 2]
```diff
@@ -114,7 +114,10 @@ struct TExportSink {
     4: required string line_delimiter
     // properties need to access broker.
     5: optional list<Types.TNetworkAddress> broker_addresses
-    6: optional map<string, string> properties;
+    6: optional map<string, string> properties
+    7: optional list<list<string>> schema
+    8: optional map<string, string> file_properties
```
Comment (morningman) [inline gensrc/thrift/DataSinks.thrift:119]: `file_properties` not used?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/LoadStmt.java (hunk 1 above)
Comment (xinghuayu007) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/LoadStmt.java]: deleted it

File: gensrc/thrift/DataSinks.thrift (hunk 2 above)
Comment (xinghuayu007) [reply gensrc/thrift/DataSinks.thrift:119]: file_properties is used to represent parquet file properties, like version, compression and so on.

**Pull request closed without merging**